libs/partners/pinecone/langchain_pinecone/embeddings.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import os
+from typing import Dict, Iterable, List, Optional
+
+import aiohttp
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import (
+    BaseModel,
+    Extra,
+    Field,
+    SecretStr,
+    root_validator,
+)
+from langchain_core.utils import convert_to_secret_str
+from pinecone import Pinecone as PineconeClient  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_BATCH_SIZE = 64
+
+
+class PineconeEmbeddings(BaseModel, Embeddings):
```

Comment:
should we mark this @beta while this is still not included in the official pinecone sdk?

Reply:
is the issue that the pinecone sdk doesn't support async natively yet?

Reply:
Correct we don't support async in the sdk didn't have an immediate plan too. Embeddings is in our latest stable API version despite being public preview for the purpose of production readiness the api will not change.

Reply:
correct